### ISSUE TITLE
Where is my TabSessionManager - Backup folder?

### ISSUE THREAD

**Issue by russherman** [sienori/Tab-Session-Manager#259]
I run more than 1 FF via a firefox.ini file. Each of them asks for the directory for each download, so there is no default Download Folder. there's no file in my userspace named "TabSessionManager - Backup" as seen my storage.js file. Ideally, However, I **can**  save my tab sessions OK, but I have no clue were they're going. I'd like to be able to specify a complete path for session backup in each of my FF instances. I'm OK with patching my js files by hand just to get that working, but the ability to change it via the Settings page would be nicer.

**Comment by sienori** [comment]
![backup](https://user-images.githubusercontent.com/29514424/40883941-c69b5b3a-6743-11e8-97f1-70e7aa454090.jpg)
Please click "Download folder" of backup option. Is the download folder displayed?

**Comment by kunal999** [comment]
Interestingly, when I open that folder, I don't see any "TabSessionManager" folder.
I am using Win 10 and I have "Always ask you where to save files" option selected in FF Downloads options.

**Comment by russherman** [comment]
My experience is the sames as kunal999 I think. The download folder shows as $HOME/Downloads. But I don't see any "TabSessionManager" folder either. 

I'm running FF 60.0.2 on Ubuntu Mate 18.04, and also using the Always ask option. That may have something to do with what we're both (not) seeing. @sienori - How do you decide where to drop the TabSessionManager folder?

**Comment by nollinvoyd** [comment]
I was thoroughly confused by this Save Backup/Save Location Feature until I read this thread.  I thought that "Download Folder" link was there to give me the power to select the download location of my choice.

When I set up a firefox profile, I first select the "Save File To" location, after which I uncheck that option, and then select "Always ask you where to save files".

Even tho the "Save Files to" location is greyed out, that is going to be the backup folder for Tab Session Manager.  The next time you launch the browser, you should see the folder in that location.

**Comment by jmm213** [comment]
This was reported June 3, 2018 with a picture of Backup - Save and a "Download Folder" option.
I have the latest  as of 9-11-2019 and NO Download Folder selection option available on FF 68.0.1

**Comment by nollinvoyd** [comment]
Here's a visual of what i described earlier.  It works.

First, discover/establish what your Firefox download directory location.  You nay opt for a custom location by browsing to and selecting a folder of your choosing. 

![TSM1](https://user-images.githubusercontent.com/20844322/62969886-14fe6700-bdc3-11e9-8633-a4b6713e3888.png)

I then select "ask me every time", because I don't always necessarily want a given file to be automatically downloaded to the default Firefox download folder.  But that's just me.  It does not hinder the formation of the "TabSessionManager - Backup folder".

![TSM2](https://user-images.githubusercontent.com/20844322/62970788-1cbf0b00-bdc5-11e9-8ff2-5a2992204dc7.png)

Bottom line is that the "TabSessionManager - Backup folder" can only be located inside the default/designated Firefox download directory displayed in your Firefox settings.

As it'd stated in the settings description "Specify the folder to save the backup file. The save location is limited to the folder under browser download folder."

Also, as far as i can see, the is no longer that "download folder" in settings link that existed in the past.

**Comment by jimsy3** [comment]
where is it stored now then and which file format was it, so that i could atleast try and search for it?
